src/services/accountService.ts

```diff
@@ -0,0 +1,148 @@
+import { BadRequestException, Injectable, NotFoundException, forwardRef } from '@nestjs/common';
+import { CurrentAccount, CurrentAccountDTO } from '../models/modelCurrentAccount';
+import { SavingsAccount, SavingsAccountDTO } from '../models/modelSavingsAccount';
+import { CustomerService } from './customerService';
+import { Inject } from '@nestjs/common';
+import { Account } from 'src/models/modelAccount';
+import { Customer } from 'src/models/modelCustomer';
+
+@Injectable()
+export class AccountService {
+  private accounts: (CurrentAccount | SavingsAccount)[] = [];
+
+  constructor(
+    @Inject(forwardRef(() => CustomerService))
+    private readonly customerService: CustomerService,
+  ) {}
+
+  createAccount(customerId: string, type: 'CURRENT' | 'SAVINGS', interestRate?: number, initialBalance: number = 0): CurrentAccountDTO | SavingsAccountDTO {
+    const customer = this.customerService.findCustomerById(customerId);
+    if (!customer) {
+      throw new NotFoundException('Customer not found.');
+    }
+  
+    let newAccount: CurrentAccount | SavingsAccount;
+  
+    if (type === 'CURRENT') {
+      if (customer.salaryIncome >= 500) {
+        const accountNumber = this.generateRandomAccountNumber();
+        const currentAccountNumber = `CA-${accountNumber}`
+        newAccount = new CurrentAccount(currentAccountNumber, initialBalance, customer);
+        newAccount.overdraftLimit = 100;
+      } else {
+        throw new BadRequestException('The customer does not meet the minimum requirements to open a current account.');
+      }
```

Comment:
Como oportunidade de quebrar seu código em mais funções, você pode criar uma função realizar a operação quando for conta corrente e outra função quando for conta poupança: `createCurrentAccount` e `createSavingsAccount`, respectivamente.

Com essa abordagem, além de quebrar o código em funções menores, você também vai evitar o uso de ifs aninhados!